client/core/core.go

```diff
@@ -4911,7 +4935,7 @@ func handlePreimageRequest(c *Core, dc *dexConnection, msg *msgjson.Message) err
 		go func() {
 			// Order request success OR fail closes the channel.
 			<-commitSig
-			if err := processPreimageRequest(c, dc, msg.ID, oid); err != nil {
+			if err := processPreimageRequest(c, dc, msg.ID, oid, req.CommitChecksum); err != nil {
```

Comment:
I just tried irl and it seems passing a nil `req.CommitChecksum` is fine atm. Maybe we want to error (or just warn for now) before this if the check sum is nil or wrong length? I'm guess up around the check for `req.OrderID` length.